From 1ec060c5ccbd16b3cc25b78c84b7d83aaf87f10a Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 10 Jul 2020 18:19:10 -0700
Subject: [PATCH 1433/1851] staging: apf: remove uio related codes

commit 18af093ff69cfbbff85652b86600be72e9dd430b from
https://github.com/Xilinx/linux-xlnx.git

The 'io_ptr' is never set. Remove it along with uio related codes.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/apf/xlnk.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/drivers/staging/apf/xlnk.c b/drivers/staging/apf/xlnk.c
index 21723ddb0077..a359c37dfaf7 100644
--- a/drivers/staging/apf/xlnk.c
+++ b/drivers/staging/apf/xlnk.c
@@ -39,7 +39,6 @@
 #include <linux/slab.h>
 #include <linux/types.h>
 #include <linux/uaccess.h>
-#include <linux/uio_driver.h>
 #include <linux/wait.h>
 
 #include "xlnk-ioctl.h"
@@ -90,7 +89,6 @@ struct xlnk_device_pack {
 	char name[64];
 	struct platform_device pdev;
 	struct resource res[8];
-	struct uio_info *io_ptr;
 	int refs;
 
 #ifdef CONFIG_XILINX_DMA_APF
@@ -180,12 +178,7 @@ static void xlnk_devpacks_free_all(void)
 	for (i = 0; i < MAX_XLNK_DMAS; i++) {
 		devpack = xlnk_devpacks[i];
 		if (devpack) {
-			if (devpack->io_ptr) {
-				uio_unregister_device(devpack->io_ptr);
-				kfree(devpack->io_ptr);
-			} else {
-				platform_device_unregister(&devpack->pdev);
-			}
+			platform_device_unregister(&devpack->pdev);
 			xlnk_devpacks_delete(devpack);
 			kfree(devpack);
 		}
@@ -315,7 +308,6 @@ static int xlnk_devregister(char *name,
 			pr_err("Failed to allocate device %s\n", name);
 			return -ENOMEM;
 		}
-		devpack->io_ptr = NULL;
 		strcpy(devpack->name, name);
 		devpack->pdev.name = devpack->name;
 
@@ -392,8 +384,6 @@ static int xlnk_dmaregister(char *name,
 		strcpy(devpack->name, name);
 		devpack->pdev.name = "xilinx-axidma";
 
-		devpack->io_ptr = NULL;
-
 		devpack->dma_chan_cfg[0].include_dre = chan0_include_dre;
 		devpack->dma_chan_cfg[0].datawidth = chan0_data_width;
 		devpack->dma_chan_cfg[0].irq = chan0_irq;
-- 
2.31.1

